### ISSUE TITLE
Representation & Training loss parsing

### ISSUE THREAD

**Issue by jotix16** [martius-lab/hitchhiking-rotations#1]
``` python
import numpy as np
from enum import Enum


def default(x):
    return x

def svd_orthonormalization(x):
    pass

def gram_schmidt_orthonormalization(x):
    pass

def quat_to_rotmat(x):
    # normalize first
    pass

def axis_angle_to_rotmat(x):
    pass

def euler_to_rotmat(x):
    pass

def quat_aug_dataset(quats: np.ndarray):
    # quats: (N, M, .., 4)
    # TODO: have to duplicate  other inputs
    return np.concatenate((quats, -quats), axis=0)

def quat_hm_dataset(quats: np.ndarray):
    # quats: (N, M, .., 4)
    return np.where(quats[..., 3] < 0, -quats, quats)


def geodesic_loss(Rot_gt, Rot_pred):
    pass

def chordial_loss(Rot_gt, Rot_pred):
    pass

class Representation(Enum):
    R9 =            {"pre_process_ds": default, "f_out": svd_orthonormalization}
    R6 =            {"pre_process_ds": default, "f_out": gram_schmidt_orthonormalization}
    QUAT =          {"pre_process_ds": default, "f_out": default}
    QUAT_AUG =      {"pre_process_ds": quat_aug_dataset, "f_out": default}
    QUAT_HM =       {"pre_process_ds": quat_hm_dataset, "f_out": default}
    AXIS_ANGLE =    {"pre_process_ds": default, "f_out": default}
    EULER =         {"pre_process_ds": default, "f_out": default}


class TrainingLoss(Enum):
    L1 = "l1"  # mean absolute error
    L2 = "l2"  # mean squared error
    QUAT_DP = "l_quat_dp"
    QUAT_CP = "l_quat_cp"
    # ------ todo ------
    CHORDIAL = chordial_loss
    GEODESIC = geodesic_loss


def get_loss(loss: TrainingLoss, representation: Representation):

    if representation in [Representation.R9, Representation.R6]:
        assert loss not in [TrainingLoss.Quat_DP, TrainingLoss.Quat_CP], f"Loss {loss} not supported for representation {representation}"
        if loss == TrainingLoss.CHORDIAL:
            print(f"Cordial loss for rotations is the same as L2 loss. Using default L2 loss")
            loss = TrainingLoss.L2

    if representation in [Representation.QUAT, Representation.QUAT_AUG, Representation.QUAT_HM]:
        if loss == TrainingLoss.CHORDIAL or loss == TrainingLoss.GEODESIC:
            representation["f_out"] = quat_to_rotmat

    if representation == Representation.AXIS_ANGLE:
        if loss == TrainingLoss.CHORDIAL or loss == TrainingLoss.GEODESIC:
            representation["f_out"] = axis_angle_to_rotmat

    if representation == Representation.EULER:
        if loss == TrainingLoss.CHORDIAL or loss == TrainingLoss.GEODESIC:
            representation["f_out"] = euler_to_rotmat

    f_out = representation["f_out"]

    train_loss = lambda y_gt, y_pred: loss(f_out(y_gt), f_out(y_pred))
    return train_loss


```
